File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum=INT_MIN;
        int check=0;
        
        for(int i=0;i<nums.size();i++)
        {
            check=check+nums[i];
            if(sum<check)
                sum=check;
            
            if(check<0)
                check=0;
            
            
        }
        
        return sum;
        
    }
};

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int ans=0;
        int max=Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
            ans=ans+nums[i];
            if(ans>max)
                max=ans;
            if(ans<=0)
                ans=0;
        }
        return max;
    }
}